src/Rules/Between.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spotwilliams\Validator\Rules;
+
+use Spotwilliams\Validator\Exceptions\ValidationFailed;
+use Spotwilliams\Validator\Contracts\ValidationRule;
+
+class Between implements ValidationRule
+{
+    private $field;
+    private $value;
+    private $constraint;
+
+    /** @inheritDoc */
+    public function apply(string $ruleName, string $field, mixed $value = null, $constraint = []): bool
+    {
+        $this->field = $field;
+        $this->value = $value;
+        $this->constraint = $constraint;
+
+        $this->isValidFieldAndConstraints();
+
+        return true;
+    }
+
+    /**
+     * Validates the given values on arguments.
+     *
+     * @throws ValidationFailed If validation fails.
+     */
+
+    private function isValidFieldAndConstraints() : void
+    {
+        if(!empty($this->constraint) && is_array($this->constraint)) {
+            if(!array_key_exists("min", $this->constraint) || !array_key_exists("max", $this->constraint)) {
+                throw new ValidationFailed("The field `{$this->field}` validation requires a min & max constraint.");
+            }
+        } else {
+            throw new ValidationFailed("The field `{$this->field}` validation requires a min & max constraint.");
+        }
```

Comment:
Try to avoid code duplication here. You are repeating the throw. 

Maybe the workflow could be:

`if(!is_array()) throw`

`if(!isset($constraint['max']) || !isset($constraint['min'])) throw`

and so on